calico_kubernetes/calico_kubernetes.py

```diff
@@ -319,23 +347,120 @@ def _apply_tags(self, profile_name, pod):
         :return:
         """
         print('Applying tags')
-        try:
-            labels = pod['metadata']['labels']
-        except KeyError:
-            # If there are no labels, there's no more work to do.
-            print('No labels found in pod %s' % pod)
-            return
-
-        for k, v in labels.iteritems():
-            tag = '%s_%s' % (k, v)
-            tag = tag.replace('/', '_')
-            print('Adding tag ' + tag)
+
+        # Grab namespace and create a tag if it exists.
+        namespace, ns_tag = self._get_namespace_and_tag(pod)
+
+        if ns_tag:
             try:
-                self.calicoctl('profile', profile_name, 'tag', 'add', tag)
+                print('Adding tag %s' % ns_tag) 
+                self.calicoctl('profile', profile_name, 'tag', 'add', ns_tag)
             except sh.ErrorReturnCode as e:
-                print('Could not create tag %s.\n%s' % (tag, e))
+                print('Could not create tag %s.\n%s' % (ns_tag, e))
+
+        # Create tags from labels
+        labels = self._get_metadata(pod, 'labels')
+        if labels:
+            for k, v in labels.iteritems():
+                tag = self._label_to_tag(k, v, namespace)
+                print('Adding tag ' + tag)
+                try:
+                    self.calicoctl('profile', profile_name, 'tag', 'add', tag)
+                except sh.ErrorReturnCode as e:
+                    print('Could not create tag %s.\n%s' % (tag, e))
+
         print('Finished applying tags.')
 
+    def _get_metadata(self, pod, key):
+        """
+        Return Metadata[key] Object given Pod
+        Returns None if no key-value exists
+        """
+        try:
+            val = pod['metadata'][key]
+            print("%s of pod %s:\n%s" % (key, pod, val))
+            return val
+        except KeyError:
+            print('No %s found in pod %s' % (key, pod))
+            return None
+
+    def _escape_chars(self, tag):
```

Comment:
nitpicky - this is supposed to be generic function, so instead of naming the argument 'tag', maybe make it 'string' or 'unescaped'